src/consumption.cpp

```diff
@@ -724,6 +724,17 @@ void player::consume_effects( item &food, bool rotten )
     if( food.has_flag( "HOT" ) && food.has_flag( "EATEN_HOT" ) ) {
         add_morale( MORALE_FOOD_HOT, 5, 10 );
     }
+    bool carnivore_food = ( has_trait( "CARNIVORE" ) && food.has_flag( "ALLERGEN_MEAT" ) &&
+                            !food.has_any_flag( carnivore_blacklist ) );
+    //Food is made out of meat (use ALLERGEN_MEAT, not CARNIVORE_OK (the latter has eggs and milk),
+    //player has carnivore mutation, and there are no non carnivore parts in the food.
+    //This also negates bad morale penalties on food made only of meat, as a carnivore.
+    if( carnivore_food && food.has_flag( "CANNIBALISM" ) && !has_trait( "CANNIBAL" ) ) {
+        carnivore_food = false; //Don't give the bonus if it is human flesh, and the player is no cannibal.
+    }
+    if( carnivore_food ) {
+        add_morale( MORALE_FOOD_CARNIVORE, 5, 20 );
```

Comment:
Instead of adding a new morale type, it would be better to just boost the morale bonus for all food that is pure meat. This would scale better.